java/src/jmri/jmrit/display/IndicatorTrackIcon.java

```diff
@@ -276,16 +284,12 @@ public void propertyChange(java.beans.PropertyChangeEvent evt) {
     private void setStatus(OBlock block, int state) {
         _status = _pathUtil.getStatus(block, state);
         if ((state & (OBlock.OCCUPIED | OBlock.RUNNING)) != 0) {
-            // It is rather unpleasant that the following needs to be done in a try-catch, but exceptions have been observed
-            try {
+            // _pathUtil.setLocoIcon must run on GUI. LocoLabel ctor causes editor to draw a graphic
+            ThreadingUtil.runOnLayoutEventually(() -> {
```

Comment:
It would be better to use `ThreadingUtil.runOnGUIEventually()`. The layout thread is currently the same as the GUI thread, but in a distant future, they could be separated.